### DISCUSSION TITLE
Basic world map

### DISCUSSION THREAD

**Discussion by jarek556** [systemed/tilemaker#710] [General]
I'd like to create generic world map with costline and country borders.
First attempt (with default config.json) gave me 24GB. I'm sure, that most of the content comes from coastline. How can I simplify coastline, so the size will be more compact ?

**Comment by jarek556** [comment]
I tried to create mbtiles with simplified_water_polygons.shp but tilemaker crashes with memory fault:
Layer place (z0-14)
Layer boundary (z0-14)
Layer poi (z12-14)
Layer poi_detail (z14-14) -> poi
Layer housenumber (z14-14)
Layer waterway (z8-14)
Layer waterway_detail (z12-14) -> waterway
Layer transportation (z4-14)
Layer transportation_name (z8-14)
Layer building (z13-14)
Layer water (z6-14)
Layer ocean (z0-14) -> water
Layer water_name (z14-14)
Layer water_name_detail (z14-14) -> water_name
Layer aeroway (z11-14)
Layer aerodrome_label (z10-14)
Layer park (z11-14)
Layer landuse (z4-14)
Layer urban_areas (z4-8) -> landuse
Layer landcover (z0-14)
Layer ice_shelf (z0-9) -> landcover
Layer glacier (z2-9) -> landcover
Layer mountain_peak (z11-14)
Bounding box -180, -75, 180, 60
Reading shapefile ocean
Reading shapefile urban_areas
Unable to open landcover/ne_10m_urban_areas/ne_10m_urban_areas.shp or landcover/ne_10m_urban_areas/ne_10m_urban_areas.SHP.
Reading shapefile ice_shelf
Unable to open landcover/ne_10m_antarctic_ice_shelves_polys/ne_10m_antarctic_ice_shelves_polys.shp or landcover/ne_10m_antarctic_ice_shelves_polys/ne_10m_antarctic_ice_shelves_polys.SHP.
Reading shapefile glacier
Unable to open landcover/ne_10m_glaciated_areas/ne_10m_glaciated_areas.shp or landcover/ne_10m_glaciated_areas/ne_10m_glaciated_areas.SHP.
Generated points: 0, lines: 0, polygons: 4
Reading .pbf ../generic2.osm.pbf
(Scanning for ways used in relations: 0%)           (785 ms)
Naruszenie ochrony pamięci

**Comment by systemed** [comment]
I think you'd need to write a custom config.json and process.lua for this, ideally using shapefiles only (from Natural Earth) and no OSM data. I'm afraid I haven't got the time at present to put one together but if you paste/link to your JSON and Lua files I'm sure I/others can help.

For these purposes you might not need to go all the way to z14, which would keep the filesize down.

**Comment by jarek556** [comment]
I think, that there is some problem with coastline rendering. I extracted admin_level=2 boundaries and coastline from planet OSM. 
Total size is 500MB, converted to mbtiles is 1.9GB (including landcover, without coastline). This is nice contour world map, only water fill is missing. Adding coastline (water_polygons.shp - 1.2GB) increases size of mbtiles file to 24GB(!).

**Reply by cldellow** [reply]
Can you try converting that mbtiles archive to a pmtiles one, following the `pmtiles convert` instructions here: https://github.com/protomaps/PMTiles?tab=readme-ov-file#creating-pmtiles ? If you get a savings, that will be instructive on what to do next.

The rationale for trying this: mbtiles is less efficient than pmtiles for storing ocean tiles. The mbtiles format stores a copy of every tile, even when they are exact duplicates of each other. The pmtiles format, on the other hand, can deduplicate redundant data. Ocean tiles are (or should be!) exact duplicates of each other, so this is wasteful. For a map from z0..z14, there are 358M tiles. If we say that 2/3rds of them are ocean, and they're 100 bytes each, that's 24GB right there. I am handwaving at the numbers here, but I suspect it might be in the ballpark.

If you do get a savings, you can try using tilemaker to emit pmtiles directly. It'll be a bit less efficient than `pmtiles convert` due to https://github.com/systemed/tilemaker/issues/653.

You could also try a different approach: ensure that you write a tile for all land, and _don't_ write any ocean tiles. Instead, when rendering your map, have the style draw a blue background as the initial layer. This relies on you being able to emit a tile for all land - I don't believe the stock OpenMapTiles configuration does this, so you'd have to fiddle a bit.
